openmdao/matrices/coo_matrix.py

```diff
@@ -75,17 +63,13 @@ def _build_coo(self, system):
             full_size = np.prod(shape)
             if dense:
                 if src_indices is None:
-                    if wrt_dist:
-                        delta = np.prod(info['shape'])
-                    else:
-                        delta = full_size
+                    delta = full_size
                 else:
-                    if wrt_dist:
-                        delta = info['shape'][0] * len(src_indices)
-                    else:
-                        delta = shape[0] * len(src_indices)
+                    delta = shape[0] * len(src_indices)
+
             elif rows is None:  # sparse matrix
                 delta = val.data.size
+
             else:  # list sparse format
                 delta = len(rows)
 
```

Comment:
There's a method called '_get_assembled_matrix' here and in dense_matrix.py that is no longer called and can be removed.